xrmocap/transform/keypoints3d/optim/prior_optimizer.py

```diff
@@ -0,0 +1,188 @@
+# yapf: disable
+import itertools
+import logging
+import numpy as np
+from typing import Union
+
+from xrmocap.data_structure.keypoints import Keypoints
+from xrmocap.transform.convention.keypoints_convention import (
+    get_keypoint_idxs_by_part,
+)
+from xrmocap.transform.limbs import search_limbs
+from .base_optimizer import BaseOptimizer
+
+# yapf: enable
+
+
+class PriorConstraint(BaseOptimizer):
+    """This optimization tool sets a bbox and remove 3D keypoints with large
+    movement between frames for move outside the bbox.
+
+    The bbox orientation is ignored.
+    """
+
+    def __init__(
+        self,
+        verbose: bool = True,
+        bbox_size: list = [2.0, 2.0, 2.0],
+        stand_only: bool = False,
+        ground_norm: list = [0.0, 1.0, 0.0],
+        standing_thr: float = 0.0,
+        use_limb_length: bool = False,
+        n_max_frame: int = 5,
+        diff_thr: float = 0.5,
+        update_mask: bool = False,
+        logger: Union[None, str, logging.Logger] = None,
+    ):
+        """Initialization for MovementConstrain optimizor.
+
+        Args:
+            verbose (bool, optional):
+                Whether to log info. Defaults to True.
+            bbox_size (list, optional):
+                Edge length of the bounding box.
+                Defaults to [2.0, 2.0, 2.0].
+            stand_only (bool, optional):
+                Only apply to the stand position.
+                Defaults to False.
+            ground_norm (list, optional):
+                Norm of the ground in the world coordinate.
+                This is used to approximate whether the person
+                is in the standing position.
+                Defaults to [0.0, 1.0, 0.0].
+            standing_thr (float, optional):
+                Threshold for standing position.
```

Comment:
Body center height threshold for standing position, in meters?

Reply:
Added details in [963ae97](https://github.com/openxrlab/xrmocap/pull/64/commits/963ae97873e93ed1ae810cf6194519becb2b5c90).